dhis-2/dhis-e2e-test/src/main/java/org/hisp/dhis/actions/metadata/MetadataPaginationActions.java

```diff
@@ -0,0 +1,165 @@
+        assert fields != null && !fields.isEmpty();
+        assert sort != null && !sort.isEmpty();
+        QueryParamsBuilder params = new QueryParamsBuilder()
+            .add( "filter=" + String.join( ",", filter ) )
+            .add( "fields=" + String.join( ",", fields ) )
+            .add( "order=" + String.join( ",", sort ) )
+            .add( "page=" + page )
+            .add( "pageSize=" + pageSize );
+
+        return get( "", params );
+
+    }
+
+    /**
+     * Executes a metadata request using pagination directives.
+     * Uses a default filter expression
+     *
+     * @param fields a List of String, containing the name of the fields to return
+     * @param sort a List of String, containing the sort expressions
+     * @param page the page to return
+     * @param pageSize the number of elements to return for each page
+     * @return an {@see ApiResponse} object
+     */
+    public ApiResponse getPaginated( List<String> fields, List<String> sort, int page, int pageSize )
+    {
+        return getPaginated( toParamList( DEFAULT_METADATA_FILTER ), fields, sort, page, pageSize );
+    }
+
+    /**
+     * Executes a metadata request using pagination directives.
+     * Uses a default filter and sort expression
+     *
+     * @param fields a List of String, containing the name of the fields to return
+     * @param page the page to return
+     * @param pageSize the number of elements to return for each page
+     * @return an {@see ApiResponse} object
+     */
+    public ApiResponse getPaginated( List<String> fields, int page, int pageSize )
+    {
+        return getPaginated( toParamList( DEFAULT_METADATA_FILTER ), fields, toParamList( DEFAULT_METADATA_SORT ), page,
+            pageSize );
+    }
+
+    /**
+     * Executes a metadata request using pagination directives.
+     * Uses a default filter, fields and sort expression
+     *
+     * @param page the page to return
+     * @param pageSize the number of elements to return for each page
+     * @return an {@see ApiResponse} object
+     */
+    public ApiResponse getPaginated( int page, int pageSize )
+    {
+        return getPaginated( toParamList( DEFAULT_METADATA_FILTER ), toParamList( DEFAULT_METADATA_FIELDS ),
+            toParamList( DEFAULT_METADATA_SORT ), page, pageSize );
+    }
+
+    /**
+     * Assert on the pagination ("pager") data within the API response
+     *
+     * @param response an {@see ApiResponse} object
+     * @param expectedTotal the expected minimum total number of metadata items
+     * @param expectedPageCount the expected minimum total number of pages
+     * @param expectedPageSize the expected value for page size
+     * @param expectedPage the expected value for the page
+     */
+    public void assertPagination( ApiResponse response, int expectedTotal, int expectedPageCount, int expectedPageSize,
+        int expectedPage )
+    {
+        JsonObject pager = response.extractJsonObject( "pager" );
```

Comment:
To ensure that report upon test failure is useful, I would use rest assure assertions instead of jUnit here. For example: 

`response.validate()
    .root("pager")
    .body("", is(notNullValue()))
    .body("pageCount", greaterThan(expectedPageCount))
    .....`
If this is used, we will get full request and response printed with assertion error and will be able to better evaluate what went wrong.

Reply:
Ah, good point, I'll change it to use RestAssure assertions.